inc/namespace.php

```diff
@@ -745,16 +875,43 @@ function analyse_ab_test_results( array $aggregations, string $test_id, int $pos
 	// Find if a variant is winning, ie. reject null hypothesis.
 	if ( $winning !== false ) {
 		$winning_variant = $variants[ $winning ];
+		// Require 99% certainty.
 		if ( ! is_null( $winning_variant['p'] ) && $winning_variant['p'] < 0.01 ) {
 			$winner = $winning;
 
 			// Pause the test.
 			update_is_ab_test_paused_for_post( $test_id, $post_id, true );
 
+			// Update the end date.
+			update_ab_test_end_time_for_post( $test_id, $post_id, milliseconds() );
+
+			// Store results safely.
+			save_ab_test_results_for_post( $test_id, $post_id, [
+				'winner' => $winner,
+				'variants' => $variants,
+			] );
+
 			/**
 			 * Dispatch action when winner found.
+			 *
+			 * @param string $test_id
+			 * @param int $post_id
+			 */
+			do_action( 'altis.experiments.test.winner_found', $test_id, $post_id );
+
+			/**
+			 * Dispatch action when winner found for test.
+			 *
+			 * @param int $post_id
 			 */
-			do_action( 'altis.ab_tests.winner_found', $test_id, $post_id );
+			do_action( "altis.experiments.test.winner_found.{$test_id}", $post_id );
+
+			/**
+			 * Run winner callback.
+			 */
+			if ( isset( $test['winner_callback'] ) && is_callable( $test['winner_callback'] ) ) {
+				call_user_func_array( $test['winner_callback'], [ $winning_variant['value'], $post_id ] );
```

Comment:
I wanted to ensure it definitely ran and also ran after the above hooks, especially if it results in modifying the post. Might be confusing to get an email about "post x" when the original title was "post y" for example.

Reply:
I'm thinking it should be possible to implement `winner_callback` by using the above hook in any case, so that the hook is useful. If we can actually use the hook itself, that helps us to dogfood it.